java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaHttpRequestRetryHandler.java

```diff
@@ -68,10 +68,14 @@ public boolean retryRequest(final IOException exception,
             return false;
         }
 
-        if (logger.isDebugEnabled() && executionCount <= getRetryCount()) {
-            String msg = String.format("Request failed, %d/%d retry.",
-                    executionCount, getRetryCount());
-            logger.debug(msg, exception);
+        if (logger.isDebugEnabled()) {
+            if (NON_RETRIABLE.contains(exception)) {
+                logger.debug("Request failed, unable to retry.", exception);
+            } else if (executionCount <= getRetryCount()) {
+                String msg = String.format("Request failed, %d/%d retry.",
+                        executionCount, getRetryCount());
+                logger.debug(msg, exception);
+            }
         }
 
         return super.retryRequest(exception, executionCount, context);
```

Comment:
Why not stash the value of the `super` call in a variable and use that to determine what to log?

That would factor in `NON_RETRIABLE` and other HttpClient behavior that might lead to this method cancelling the retry.